crates/stark-backend/src/circuit_api.rs

```diff
@@ -1,23 +0,0 @@
-use async_trait::async_trait;
-
-use crate::{
-    config::StarkGenericConfig,
-    prover::types::{Proof, ProofInput},
-    verifier::VerificationError,
-};
-
-/// Async prover for a specific circuit using a specific Stark config.
-#[async_trait]
-pub trait AsyncCircuitProver<SC: StarkGenericConfig> {
```

Comment:
@nyunyunyunyu I deleted the file because I'm not sure `ProofInput` will be a canonical structure always used, and also this trait implicits that the trace matrix is a solid boundary for prover input, whereas as we've discussed the format of trace may differ depending on the prover/hardware.

I was thinking we can move this into the sdk for now?